wezterm-mux-server-impl/src/sessionhandler.rs

```diff
@@ -945,6 +945,26 @@ impl SessionHandler {
                 .detach();
             }
 
+            Pdu::TogglePaneZoomState(TogglePaneZoomState { pane_id }) => {
+                spawn_into_main_thread(async move {
+                    catch(
+                        move || {
+                            let mux = Mux::get();
+                            let (_domain_id, _window_id, tab_id) = mux
+                                .resolve_pane_id(pane_id)
+                                .ok_or_else(|| anyhow!("no such pane {}", pane_id))?;
+                            let tab = mux
+                                .get_tab(tab_id)
+                                .ok_or_else(|| anyhow!("no such tab {}", tab_id))?;
+                            tab.toggle_zoom();
```

Comment:
Thanks for diving in!

There's a nuanced difference between the behavior listed in the CLI and what is actually happening here.

The CLI logic takes a pane_id parameter (defaulting to the pane from the environment), but rather than manipulating the zoom status of that pane, it toggles whichever pane is zoomed in its containing tab, rather than the requested pane.

Note that there is already a `SetPaneZoomed` PDU in the mux protocol; it also happens to have this same scoping issue.

What I think the forward should be is:

1. Fix `SetPaneZoomed` to explicitly check the zoomed status of the pane.  To support this, add a method to `Tab` and `TabInner` called `get_zoomed_pane` that returns `self.zoomed.clone()`.  Then the code can use `tab.set_zoomed(false)` to unset the zoom state directly, and then, if zooming the target pane, `tab.set_active_pane()` to activate the target tab and `tab.set_zoomed(true)` to make it zoomed.

2. With that operation fixed, the rest of this PR doesn't need to modify the codec; all the logic can reside in the CLI.
 
For the CLI I think it would be better overall if we provided a choice of three actions:
  * Explicitly set the zoom state to either true or false.
  * To toggle the zoom state
 
eg: name the subcommand `zoom-pane` and give it parameters: `--zoom`, `--unzoom`, and `--toggle` to select the behavior.  I would lean towards `--zoom` being the default because that feels like the most likely action for `zoom-pane` to take by default.

The zoom and unzoom cases are straightforward and can call `client.set_zoomed` and pass the parameters through.

For the toggle case, it's a bit more involved:

* Use `client.list_panes()` to obtain tab and pane information (see `move_pane_to_new_tab.rs`)
* Resolve the tab_id of the target tab from that information
* Resolve the pane_id that is active + zoomed
* If the active + zoomed pane_id matches the target pane, then use `client.set_zoomed` to set zoomed to false for the target pane
* Otherwise, use `client.set_zoomed` to set zoomed to true for the target pane.

Why not add a `TogglePaneZoomState` PDU and handle this on the server side? Such a toggle action would aggregate multiple functions which would make the mux protocol more complex overall.  Having discrete, composable PDUs make things simpler and more flexible in the long run.

Reply:
I've just made the changes based on your suggestion. Please review. Thank you.